ddtrace/tracer/tracer.go

```diff
@@ -190,6 +194,22 @@ func newTracer(opts ...StartOption) *tracer {
 	return t
 }
 
+// Flush flushes any buffered traces. Flush is in effect only if a tracer
+// is started. Users do not have to call Flush in order to ensure that
+// traces reach Datadog. It is a convenience method dedicated to a specific
+// use case described below.
+//
+// Flush is of use in Lambda environments, where starting and stopping
+// the tracer on each invokation may create too much latency. In this
+// scenario, a tracer may be started and stopped by the parent process
+// whereas the invokation can make use of Flush to ensure any created spans
+// reach the agent.
+func Flush() {
+	if t, ok := internal.GetGlobalTracer().(*tracer); ok {
+		t.flush <- struct{}{}
+	}
+}
```

Comment:
I think there is a potential race condition here. When `Flush()` returns to the caller, it doesn't guarantee that the flush operation has actually finished. In fact it may not even have started yet. The only guarantee is that the `flush` message was received by the `worker` goroutine.

Perhaps it would be good to change the type of the `flush` channel to `chan chan struct{}` (or similar) so that the worker can send a message back after the flush has succeeded.

Reply:
You're absolutely right, that's the correct approach. We want this to be synchronous. Created https://github.com/DataDog/dd-trace-go/pull/901